packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -58,6 +66,8 @@ export const onEvent = async (event: AWSCDKAsyncCustomResource.OnEventRequest):
       if (!describeTableResult.Table) {
         throw new Error(`Could not find ${event.PhysicalResourceId} to update`);
       }
+      log('Current table state: ', describeTableResult);
```

Comment:
Commenting here since I can't comment on unchanged `onEvent` declaration: There's an awful lot going on in this method. For a future PR, we should refactor this to simplify the flow through the event handler; to clearly separate items that can be done with an "update table" vs. things that must be done async (GSI updates). Doesn't block this commit, but we should clean it up before GA.

Reply:
+1 to generally breaking this apart a bit once we've got pentest unblocked.